utils/v2/connector/impl/DataPackageUtils.ts

```diff
@@ -0,0 +1,91 @@
+import EvmPriceSigner from "redstone-node/dist/src/signers/EvmPriceSigner";
+import _ from "lodash";
+import { PriceDataType, SignedPriceDataType } from "../PriceFeedConnector";
+import { SignedDataPackageResponse } from "./fetchers/Fetcher";
+import { PriceFeedOptions, ValueSelectionAlgorithm } from "./RedStonePriceFeed";
+
+const priceSigner = new EvmPriceSigner();
+
+export function selectDataPackage(
+  packages: SignedDataPackageResponse[],
+  valueSelectionAlgorithm: ValueSelectionAlgorithm,
+): SignedDataPackageResponse {
+  const sortedPackages = [...packages];
+  sortedPackages.sort((p1, p2) => p1.timestamp - p2.timestamp); // sorting prices from oldest to newest
+  switch (valueSelectionAlgorithm) {
+
+    // TODO: improve the implementation for first-valid
+    // We don't really need to wait for all responses within
+    // the first-valid method. It's enough to receive the first
+    // valid answer
+    case "first-valid":
+      return packages[packages.length - 1];
+
+    case "newest-valid":
+      return packages[packages.length - 1];
+
+    case "oldest-valid":
+      return packages[0];
+
+    default:
+      throw new Error(
+        `Unsupported value for valueSelectionAlgorithm: ${valueSelectionAlgorithm}`);
+  }
+}
+
+export function validateDataPackage(
+  fetchedPackage: SignedDataPackageResponse,
+  priceFeedOptions: PriceFeedOptions,
+  signer: string,
+): boolean {
+
+  const maxTimestampDiffMilliseconds = priceFeedOptions.dataSources?.maxTimestampDiffMilliseconds;
+  const preVerifySignatureOffchain = priceFeedOptions.dataSources?.preVerifySignatureOffchain;
+
+  // Checking timestamp diff
+  const timeDiffMilliseconds = Date.now() - fetchedPackage.timestamp;
+
+  // Checking timestamp diff
+  if (maxTimestampDiffMilliseconds && maxTimestampDiffMilliseconds < timeDiffMilliseconds) {
+    console.warn(`Timestamp is too old: ${fetchedPackage.timestamp}`);
+    return false;
+  }
+
+  // Offchain signature verification
+  // (only liteSignature offchain verification is implemented for now)
+  if (preVerifySignatureOffchain) {
+    const isValidSignature = priceSigner.verifyLiteSignature({
+      pricePackage: {
+        prices: fetchedPackage.prices,
+        timestamp: fetchedPackage.timestamp,
+      },
+      signer,
+      signature: fetchedPackage.signature,
```

Comment:
Do we need a signature for light signature verification?

Reply:
Nope, but the function receives an object of type, that must have this prop :)